File: app/src/main/java/com/gustu/logbook/fragment/home/adapter/HomeAdapter.java
package com.gustu.logbook.fragment.home.adapter;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.gustu.logbook.R;
import com.gustu.logbook.fragment.home.model.DataItem;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class HomeAdapter extends RecyclerView.Adapter<HomeAdapter.ViewHolder> {
    List<DataItem> dataItemList;
    Context context;

    public void setContext(Context context) {
        this.context = context;
    }

    public HomeAdapter(List<DataItem> dataItemList) {
        this.dataItemList = dataItemList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_home_logbook,parent,false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        DataItem dataItem = dataItemList.get(position);
        if (position%2==0){
            holder.layout.setBackgroundColor(Color.GRAY);
        }
        else{
            holder.layout.setBackgroundColor(Color.WHITE);
        }
        holder.namaKegiatan.setText(dataItem.getTLBNAMAKEGIATAN());
        holder.tanggal.setText(String.valueOf(dataItem.getTLBTANGGAL()));
    }

    @Override
    public int getItemCount() {
        if (dataItemList!=null){
            return dataItemList.size();
        }
        return 0;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.namaKegiatanLog)
        TextView namaKegiatan;
        @BindView(R.id.tanggalLog)
        TextView tanggal;
        @BindView(R.id.btEdit)
        ImageView edit;
        @BindView(R.id.btHapus)
        ImageView hapus;
        @BindView(R.id.layoutItemLogbook)
        LinearLayout layout;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            ButterKnife.bind(this,itemView);
        }
    }
}

File: app/src/main/java/com/gustu/logbook/fragment/home/adapter/HomeAdapter.kt
package com.gustu.logbook.fragment.home.adapter

import android.content.Context
import android.content.Intent
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.gustu.logbook.R
import com.gustu.logbook.fragment.home.interfaces.HomeView
import com.gustu.logbook.fragment.home.model.GetLogBook
import com.gustu.logbook.fragment.home.presenter.DeletePresenter
import com.gustu.logbook.fragment.home.presenter.HomePresenter
import com.gustu.logbook.fragment.home.view.HomeFragment
import com.gustu.logbook.main.view.MainActivity
import kotlinx.android.synthetic.main.item_home_logbook.view.*

class HomeAdapter(internal var dataItemList: List<GetLogBook>?) : RecyclerView.Adapter<HomeAdapter.ViewHolder>() {
    internal lateinit var context: Context
    internal lateinit var homeFragment: HomeFragment
    fun setContext(context: Context , homeFragment: HomeFragment) {
        this.context = context
        this.homeFragment = homeFragment
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_home_logbook, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val dataItem = dataItemList!![position]
//        if (position % 2 == 0) {
//            holder.layout!!.setBackgroundColor(Color.LTGRAY)
//        } else {
//            holder.layout!!.setBackgroundColor(Color.WHITE)
//        }
        holder.namaKegiatan!!.text = dataItem.tLBNAMAKEGIATAN!!
        holder.tanggal!!.text = dataItem.tLBTANGGAL
        holder.hapus.setOnClickListener {
            homeFragment.deleteData(dataItemList!!.get(position).tLBID!!)
        }
    }

    override fun getItemCount(): Int {
        return if (dataItemList != null) {
            dataItemList!!.size
        } else 0
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        internal var namaKegiatan = itemView.namaKegiatanLog
        internal var tanggal = itemView.tanggalLog
        internal var edit = itemView.btEdit
        internal var hapus = itemView.btHapus
        internal var layout = itemView.layoutItemLogbook
    }
}
